src/twisted/names/test/test_dns.py

```diff
@@ -427,14 +429,18 @@ def test_Charstr(self):
             self.assertEqual(result.string, n)
 
 
-    def _recordRoundtripTest(self, record):
+    def _recordRoundtripTest(self, record, expectedEncoding=None):
         """
         Assert that encoding C{record} and then decoding the resulting bytes
-        creates a record which compares equal to C{record}.
+        creates a record which compares equal to C{record}. Optionally assert
+        that the encoded bytes are equal to C{expectedEncoding}.
```

Comment:
The tests are also using pydoctor / epydoc markup for arguments. What is `C{record}` ?


But rather than adding an optional argument, I think that it would be best to have a separate/dedicated method.

I think that the name of this function can be changed to something like `_assertRecordSerialization` or _assertRecordEncoding.

The new method can do a call to _recordRoundtripTest() if required

Also, instead of encoding we can use expectedWireBytes or something like that.